src/Test/L0/Listener/Configuration/Testable/TestWindowsServiceControlManager.cs

```diff
@@ -0,0 +1,28 @@
+using System.ServiceProcess;
+
+using Microsoft.VisualStudio.Services.Agent.Listener.Configuration;
+
+namespace Test.L0.Listener.Configuration.Testable
+{
+    public class TestWindowsServiceControlManager : WindowsServiceControlManager
+    {
+        public ServiceController SelectedServiceController { get; set; }
+
+        protected override void InstallService(
```

Comment:
I would rather this leverage the service locator pattern to abstract away dependencies (or untestable code) rather than expose new virtual methods on the production classes (that are only overridden by test classes). Let me know if there is something I'm overlooking here. It looks like ultimately that's your goal, abstracting away untestable portions of the code b/c it calls into native APIs.